From 999862489c7fb8c107606beaf13920a4a73292d7 Mon Sep 17 00:00:00 2001
From: Johnny Mnemonic <jm@machine-hall.org>
Date: Fri, 22 Nov 2024 10:52:39 +0100
Subject: [PATCH 39/63] ia64: replay "asm-generic: provide generic page_to_phys
 [...]"

See c5c3238d9b8cee58cd4b08bbbe9347a94a566390.

Also fix the following fallout resulting from the upstream change:
```
  CC      lib/kobject_uevent.o
In file included from ./include/asm-generic/bug.h:22,
                 from ./arch/ia64/include/asm/bug.h:17,
                 from ./include/linux/bug.h:5,
                 from ./include/linux/thread_info.h:13,
                 from ./include/asm-generic/preempt.h:5,
                 from ./arch/ia64/include/generated/asm/preempt.h:1,
                 from ./include/linux/preempt.h:79,
                 from ./include/linux/spinlock.h:56,
                 from ./include/linux/mmzone.h:8,
                 from ./include/linux/gfp.h:7,
                 from ./include/linux/umh.h:4,
                 from ./include/linux/kmod.h:9,
                 from ./include/linux/module.h:17,
                 from drivers/char/agp/i460-agp.c:10:
drivers/char/agp/i460-agp.c: In function 'i460_insert_memory_small_io_page':
drivers/char/agp/i460-agp.c:300:18: warning: format '%lx' expects argument of type 'long unsigned int', but argument 6 has type 'long long unsigned int' [-Wformat=]
  300 |         pr_debug("i460_insert_memory_small_io_page(mem=%p, pg_start=%ld, type=%d, paddr0=0x%lx)\n",
      |                  ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
./include/linux/printk.h:391:21: note: in definition of macro 'pr_fmt'
  391 | #define pr_fmt(fmt) fmt
      |                     ^~~
./include/linux/dynamic_debug.h:248:9: note: in expansion of macro '__dynamic_func_call_cls'
  248 |         __dynamic_func_call_cls(__UNIQUE_ID(ddebug), cls, fmt, func, ##__VA_ARGS__)
      |         ^~~~~~~~~~~~~~~~~~~~~~~
./include/linux/dynamic_debug.h:250:9: note: in expansion of macro '_dynamic_func_call_cls'
  250 |         _dynamic_func_call_cls(_DPRINTK_CLASS_DFLT, fmt, func, ##__VA_ARGS__)
      |         ^~~~~~~~~~~~~~~~~~~~~~
./include/linux/dynamic_debug.h:269:9: note: in expansion of macro '_dynamic_func_call'
  269 |         _dynamic_func_call(fmt, __dynamic_pr_debug,             \
      |         ^~~~~~~~~~~~~~~~~~
./include/linux/printk.h:625:9: note: in expansion of macro 'dynamic_pr_debug'
  625 |         dynamic_pr_debug(fmt, ##__VA_ARGS__)
      |         ^~~~~~~~~~~~~~~~
drivers/char/agp/i460-agp.c:300:9: note: in expansion of macro 'pr_debug'
  300 |         pr_debug("i460_insert_memory_small_io_page(mem=%p, pg_start=%ld, type=%d, paddr0=0x%lx)\n",
      |         ^~~~~~~~
drivers/char/agp/i460-agp.c:300:94: note: format string is defined here
  300 |         pr_debug("i460_insert_memory_small_io_page(mem=%p, pg_start=%ld, type=%d, paddr0=0x%lx)\n",
      |                                                                                            ~~^
      |                                                                                              |
      |                                                                                              long unsigned int
      |                                                                                            %llx
  CC      kernel/bpf/map_iter.o
```
---
 arch/ia64/include/asm/page.h | 1 -
 drivers/char/agp/i460-agp.c  | 2 +-
 2 files changed, 1 insertion(+), 2 deletions(-)

diff --git a/arch/ia64/include/asm/page.h b/arch/ia64/include/asm/page.h
index 310b09c3342d..dc6f0f2268be 100644
--- a/arch/ia64/include/asm/page.h
+++ b/arch/ia64/include/asm/page.h
@@ -95,7 +95,6 @@ do {						\
 
 #include <asm-generic/memory_model.h>
 
-#define page_to_phys(page)	(page_to_pfn(page) << PAGE_SHIFT)
 #define virt_to_page(kaddr)	pfn_to_page(__pa(kaddr) >> PAGE_SHIFT)
 #define pfn_to_kaddr(pfn)	__va((pfn) << PAGE_SHIFT)
 
diff --git a/drivers/char/agp/i460-agp.c b/drivers/char/agp/i460-agp.c
index 15b240ea4848..a8d842749b6a 100644
--- a/drivers/char/agp/i460-agp.c
+++ b/drivers/char/agp/i460-agp.c
@@ -297,7 +297,7 @@ static int i460_insert_memory_small_io_page (struct agp_memory *mem,
 	int i, j, k, num_entries;
 	void *temp;
 
-	pr_debug("i460_insert_memory_small_io_page(mem=%p, pg_start=%ld, type=%d, paddr0=0x%lx)\n",
+	pr_debug("i460_insert_memory_small_io_page(mem=%p, pg_start=%ld, type=%d, paddr0=0x%llx)\n",
 		 mem, pg_start, type, page_to_phys(mem->pages[0]));
 
 	if (type >= AGP_USER_TYPES || mem->type >= AGP_USER_TYPES)
-- 
2.25.1

